### PR TITLE
Cleanup metadata macros

### LINKED ISSUE #432: Metadata without macros
### Feature type

Changing existing functionality

### Improvement Description

The metadata of struct fields is generated with macros. While fancy, it is not very readable/explicit (code), and it is only used for model coupling through BMI and thus not exposed to a wflow user (running a wflow model simulation). Implementing metadata without macros makes the code more readable/explicit.  

### Implementation Description

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by verseve**
## Issue addressed
Fixes #432 

## Explanation
See issue #432. The macro `get_units` has not been removed. This macro is nice to have and also used by `BMI` without affecting code readability. Note that this branch has been started from the `vertical_concepts` branch.

### CODE REVIEW

File: src/flow.jl [hunk 1]
```diff
@@ -38,29 +37,44 @@ abstract type SurfaceFlow end
...
+        1
+    end
+end
+
+function grid_location(::SurfaceFlow, var)
+    if var in (:dt, :its, :kinwave_it)
+        "none"
+    else
+        "node"
+    end
```
Comment (JoostBuitink) [inline src/flow.jl:77]: I'm not too familiar with these functions, but it feels like they are quite specific to BMI: is that right? Is it maybe better to keep track of these in the src/bmi.jl file, or do you prefer to keep it close to their source? If it is quite specific BMI information/functionality, maybe it is better to keep those functions in the scr/bmi.jl file (a bit similar to the src/states.jl functionality). If it is not specific for BMI, keeping them here is fine with me!

**Review by JoostBuitink — CHANGES_REQUESTED** [review]
Looks good to me, code is indeed a lot cleaner with these changes! I have one question/comment regarding the location of the `exchange` and `grid_location` functions, see the comment related to that!

File: src/flow.jl (hunk 1 above)
Comment (verseve) [reply src/flow.jl:77]: Yes, these functions are quite specific to BMI. I agree and will move these functions to src/bmi.jl.

**Review by JoostBuitink — APPROVED** [review]
Looks good, and nice to see CI being updated for the v1 branch as well!

**Pull request merged**